engine/pike/pike_test.go

```diff
@@ -0,0 +1,46 @@
+package pike
+
+import (
+	"encoding/hex"
+	"testing"
+
+	"github.com/libsv/go-bk/bec"
+	assert "github.com/stretchr/testify/require"
+
+	"github.com/bitcoin-sv/spv-wallet/engine/script/template"
+)
+
+func TestGenerateLockingScriptsFromTemplates(t *testing.T) {
+	// Define sample data
+	senderPubKeyHex := "027c1404c3ecb034053e6dd90bc68f7933284559c7d0763367584195a8796d9b0e"
+	senderPubKeyBytes, err := hex.DecodeString(senderPubKeyHex)
+	assert.NoError(t, err)
+	senderPubKey, err := bec.ParsePubKey(senderPubKeyBytes, bec.S256())
+	assert.NoError(t, err)
+
+	receiverPubKeyHex := "03a34e456deecb6e6e9237e63e5b7d045d1d2a456eb6be43de1ec4e9ac9a07b50d"
+	receiverPubKeyBytes, err := hex.DecodeString(receiverPubKeyHex)
+	assert.NoError(t, err)
+	receiverPubKey, err := bec.ParsePubKey(receiverPubKeyBytes, bec.S256())
+	assert.NoError(t, err)
+
+	outputsTemplate := []template.P2PKHTemplate{
+		{Script: "76a914fd88ac"}, // Example P2PKH script in hex
+		{Script: "76a914fe88ad"}, // Another example P2PKH script in hex
+	}
+
+	t.Run("Valid Case", func(t *testing.T) {
+		lockingScripts, err := GenerateLockingScriptsFromTemplates(outputsTemplate, senderPubKey, receiverPubKey, "test-reference")
```

Comment:
IMHO we should validate if lockingScripts, contains what we expect. I mean exactly those script strings that we know should be there.